packages/store/ts/config/v2/store.ts

```diff
@@ -22,44 +23,43 @@ export type validateStoreTablesConfig<input, scope extends AbiTypeScope = AbiTyp
 };
 
 export type resolveStoreTablesConfig<input, scope extends AbiTypeScope = AbiTypeScope> = evaluate<{
-  [key in keyof input]: resolveTableConfig<input[key], scope>;
+  readonly [key in keyof input]: resolveTableConfig<input[key], scope>;
 }>;
 
-type scopeWithUserTypes<userTypes, scope extends AbiTypeScope = AbiTypeScope> = UserTypes extends userTypes
+export type scopeWithUserTypes<userTypes, scope extends AbiTypeScope = AbiTypeScope> = UserTypes extends userTypes
   ? scope
   : userTypes extends UserTypes
     ? extendScope<scope, userTypes>
     : scope;
 
-type scopeWithEnums<enums, scope extends AbiTypeScope = AbiTypeScope> = Enums extends enums
+export type scopeWithEnums<enums, scope extends AbiTypeScope = AbiTypeScope> = Enums extends enums
   ? scope
   : enums extends Enums
     ? extendScope<scope, { [key in keyof enums]: "uint8" }>
     : scope;
 
+export type extendedScope<input> = scopeWithEnums<get<input, "enums">, scopeWithUserTypes<get<input, "userTypes">>>;
+
 export type validateStoreConfig<input> = {
   [key in keyof input]: key extends "tables"
-    ? validateStoreTablesConfig<
-        input[key],
-        scopeWithEnums<get<input, "enums">, scopeWithUserTypes<get<input, "userTypes">>>
-      >
+    ? validateStoreTablesConfig<input[key], extendedScope<input>>
     : key extends "userTypes"
       ? UserTypes
       : key extends "enums"
         ? narrow<input[key]>
         : input[key];
 };
 
+export type resolveEnums<enums> = { [key in keyof enums]: Readonly<enums[key]> };
```

Comment:
do we have a rule for when we use `readonly` keys vs not?

Reply:
Should probably always be `readonly`. I think the test is passing in this case because of the `const input` parameter, so the incoming enum config is already readonly, which makes the enum object on the output readonly too. Will still add it here for consistency.